src/Analyser/Scope.php

```diff
@@ -845,6 +846,21 @@ private function resolveType(Expr $node): Type
 			$leftType = $this->getType($left);
 			$rightType = $this->getType($right);
 
+			if ($node instanceof BinaryOp) {
```

Comment:
PHP extensions cannot influence how AssignOp works?

Reply:
What do you think about it? https://github.com/phpstan/phpstan/pull/2114/commits/ed6192ec93fd7bb8c3746f3b13efdf8e22cc5471

Reply:
Yes, that's nice!